### PR TITLE
Fixed publish release to be on main branch instead

### PR DESCRIPTION
**Pull request by shanejonas**
See here for more info: https://github.com/MetaMask/api-specs/pull/224/files#r1583576395

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: Publish Release
 
 on:
-  pull_request:
-    types: [closed]
+  push:
+    branches: [main]
   workflow_call:
```
Comment (Gudahtt) [inline .github/workflows/publish-release.yml:6]: I don't see any workflow_call triggers. Maybe these can be removed?

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -12,14 +12,26 @@ on:
...
+    outputs:
+      IS_RELEASE: ${{ steps.is-release.outputs.IS_RELEASE }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: MetaMask/action-is-release@v1
+        id: is-release
   publish-release:
     permissions:
       contents: write
     # The second argument to startsWith() must match the release-branch-prefix
```
Comment (Gudahtt) [inline .github/workflows/publish-release.yml:31]: Nit: This comment is no longer valid, it was referring to the filter that was removed here

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by shanejonas** [comment]
@Gudahtt addressed your comments in the last 2 commits

**Pull request merged**